### PR TITLE
ZEN-22313 Service Migration for MariaDB buffer pool default

### PR DESCRIPTION
**Pull request by carlcamera**
Test changes to MariaDB buffer pool default value.

### CODE REVIEW

File: Products/ZenModel/migrate/setMariaDbBufferPoolSize.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import Migrate
+import servicemigration as sm
+
+sm.require("1.0.0")
+
+class SetMariaDbBufferPoolSize(Migrate.Step):
+    """Setting MariaDB buffer pool size default"""
+
+    version = Migrate.Version(5, 0, 70)
```
Comment (smousa) [inline Products/ZenModel/migrate/setMariaDbBufferPoolSize.py]: This is the wrong migration version

**Comment by carlcamera** [comment]
Per QUES-33 it looks like the migration version should be (5,2,0)

File: Products/ZenModel/migrate/setMariaDbBufferPoolSize.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        marialist = ['mariadb', 'mariadb-model']
+        marias = filter(lambda s: s.name in marialist, ctx.services)
+
+        commit = False
+        chk1 = 'innodb_buffer_pool_size = {{percentScale .RAMCommitment 0.8}}'
+        chk2 = 'innodb_buffer_pool_size = 512M'
+        repl = 'innodb_buffer_pool_size = {{.CPUCommitment}}G'
+        for maria in marias:
+            for cnf in maria.configFiles:
```
Comment (jplouis) [inline Products/ZenModel/migrate/setMariaDbBufferPoolSize.py]: I believe you need to update the original config file property as well

File: Products/ZenModel/migrate/setMariaDbBufferPoolSize.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        marialist = ['mariadb', 'mariadb-model']
+        marias = filter(lambda s: s.name in marialist, ctx.services)
+
+        commit = False
+        chk1 = 'innodb_buffer_pool_size = {{percentScale .RAMCommitment 0.8}}'
+        chk2 = 'innodb_buffer_pool_size = 512M'
+        repl = 'innodb_buffer_pool_size = {{.CPUCommitment}}G'
```
Comment (jplouis) [inline Products/ZenModel/migrate/setMariaDbBufferPoolSize.py]: I don't think you want to update these if they have diverted from the original. A customer may have customized it and we want to honor their changes as much as possible.

File: Products/ZenModel/migrate/setMariaDbBufferPoolSize.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            ctx = sm.ServiceContext()
+        except sm.ServiceMigrationError:
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        marialist = ['mariadb', 'mariadb-model']
+        marias = filter(lambda s: s.name in marialist, ctx.services)
+
+        commit = False
+        chk1 = 'innodb_buffer_pool_size = {{percentScale .RAMCommitment 0.8}}'
```
Comment (jplouis) [inline Products/ZenModel/migrate/setMariaDbBufferPoolSize.py]: Leave innodb_buffer_pool_size as is.  Change the default RAMCommitment in the svc to 4G out of the box.  Change innodb_buffer_pool_instances to CPUCommitment -1 per ZEN-21973

File: Products/ZenModel/migrate/tests/test_setMariaDbBufferPoolSize.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+
+import os
+import unittest
+
+import Globals
+import common
+
+
+class Test_SetMariaDbTimeouts(unittest.TestCase, common.ServiceMigrationTestCase):
```
Comment (jplouis) [inline Products/ZenModel/migrate/tests/test_setMariaDbBufferPoolSize.py]: This seems to be testing timeout changes not buffer pool changes.

**Comment by carlcamera** [comment]
test this.

**Comment by carlcamera** [comment]
test this.

File: Products/ZenModel/migrate/setMariaDbBufferPoolSize.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    def cutover(self, dmd):
+
+        try:
+            ctx = sm.ServiceContext()
+        except sm.ServiceMigrationError:
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        marialist = ['mariadb', 'mariadb-model']
```
Comment (smousa) [inline Products/ZenModel/migrate/setMariaDbBufferPoolSize.py]: do we care about mariadb-events as well?

Comment (carlcamera) [inline Products/ZenModel/migrate/setMariaDbBufferPoolSize.py]: JP and Ian said that maria-events would have a different RAM requirement and they were happy that it was not included in the check.

**Pull request merged**